mod/node-api/backend/beacon.go

```diff
@@ -24,158 +24,190 @@ import (
 	"context"
 	"strconv"
 
-	types "github.com/berachain/beacon-kit/mod/consensus-types/pkg/types"
-	serverType "github.com/berachain/beacon-kit/mod/node-api/server/types"
+	"github.com/berachain/beacon-kit/mod/node-api/backend/storage"
+	"github.com/berachain/beacon-kit/mod/node-api/server/types"
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/common"
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/crypto"
 	"github.com/berachain/beacon-kit/mod/primitives/pkg/math"
 )
 
-func (h Backend) GetGenesis(ctx context.Context) (common.Root, error) {
+// GetGenesis returns the genesis state of the beacon chain.
+func (b Backend[
+	_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _,
+]) GetGenesis(ctx context.Context) (common.Root, error) {
 	// needs genesis_time and gensis_fork_version
```

Comment:
do we want to make a StateID -> BeaconState Middleware thing? cause this code is gonna be repeated over and over

Reply:
thats kinda my idea with storage/backend but i didnt really know where to put that